project/src/test/puzzle/test-rank-calculator.gd

```diff
@@ -356,3 +356,20 @@ func test_unranked_loss() -> void:
 	var rank := _rank_calculator.calculate_rank()
 	assert_eq(RankCalculator.grade(rank.seconds_rank), "-")
 	assert_eq(RankCalculator.grade(rank.score_rank), "-")
+
+
+func test_extra_seconds_per_piece() -> void:
+	CurrentLevel.settings.rank.extra_seconds_per_piece = 1.2
+	CurrentLevel.settings.set_finish_condition(Milestone.LINES, 300)
+	
+	PuzzleScore.level_performance.seconds = 240
+	PuzzleScore.level_performance.lines = 60
+	PuzzleScore.level_performance.box_score = 600
+	PuzzleScore.level_performance.combo_score = 500
+	PuzzleScore.level_performance.score = 1160
+	var rank := _rank_calculator.calculate_rank()
+	assert_eq(RankCalculator.grade(rank.speed_rank), "M")
+	assert_eq(RankCalculator.grade(rank.lines_rank), "A+")
+	assert_eq(RankCalculator.grade(rank.box_score_per_line_rank), "S+")
+	assert_eq(RankCalculator.grade(rank.combo_score_per_line_rank), "S-")
+	assert_eq(RankCalculator.grade(rank.score_rank), "AA+")
```

Comment:
Remove these other non-speed assertions

Reply:
Improved test to more specifically test two scenarios, and only `speed_rank` and `score_rank`